server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -671,14 +671,17 @@ public Engine.IndexResult applyIndexOperationOnPrimary(ParsedDocument parsedDoc,
                                                            boolean isRetry) throws IOException {
         assert VersionType.INTERNAL.validateVersionForWrites(version);
         Term uid = new Term(IdFieldMapper.NAME, Uid.encodeId(parsedDoc.id()));
+        var origin = Engine.Operation.Origin.PRIMARY;
+        ensureWriteAllowed(origin);
+
```

Comment:
We accidentally missed this since in 5.2/legacyInsert we used `applyIndexOperationOnPrimary` which had such check via calling `applyIndexOperation`.

See https://github.com/crate/crate/commit/672a882a82a442cdb007fdfe7eac9391d17cf5e6#diff-afc8b63b41c53d6e6ea5b956bdd39557ab55e1dfe1b3e1242f1d4e29a9fb57a0L640

I thought that for primary `ensureWriteAllowed` call may be not that valuable as it's mainly about  
assert assertPrimaryMode();

However there is also `WRITE_ALLOWED_STATES` check

Reply:
The check happens here: 

https://github.com/crate/crate/blob/008ea43fe1db7da38d1d00bd486f97005c01092a/server/src/main/java/org/elasticsearch/index/shard/IndexShard.java#L776

which is called here:

https://github.com/crate/crate/blob/008ea43fe1db7da38d1d00bd486f97005c01092a/server/src/main/java/io/crate/execution/dml/upsert/TransportShardUpsertAction.java#L624

This change is not needed

Reply:
I see, will remove